packages/logger/src/logger.injectable.ts

```diff
@@ -12,18 +12,66 @@ export const loggerInjectionToken = getInjectionToken<Logger>({
 
 export const loggerInjectable = getInjectable({
   id: "logger",
-  instantiate: (di): Logger => {
-    const baseLogger = di.inject(winstonLoggerInjectable);
-
-    return {
-      debug: (message, ...data) => baseLogger.debug(message, ...data),
-      info: (message, ...data) => baseLogger.info(message, ...data),
-      warn: (message, ...data) => baseLogger.warn(message, ...data),
-      error: (message, ...data) => baseLogger.error(message, ...data),
-      silly: (message, ...data) => baseLogger.silly(message, ...data),
-    };
-  },
+  instantiate: (di): Logger => ({
+    debug: di.inject(logDebugInjectionToken),
+    info: di.inject(logInfoInjectionToken),
+    warn: di.inject(logWarningInjectionToken),
+    error: di.inject(logErrorInjectionToken),
+    silly: di.inject(logSillyInjectionToken),
+  }),
 
   decorable: false,
   injectionToken: loggerInjectionToken,
 });
+
+export type LogFunction = (message: string, ...data: any[]) => void;
+
+export const logDebugInjectionToken = getInjectionToken<LogFunction>({
+  id: "log-debug-injection-token",
+});
+
+export const logInfoInjectionToken = getInjectionToken<LogFunction>({
+  id: "log-info-injection-token",
+});
+
+export const logWarningInjectionToken = getInjectionToken<LogFunction>({
+  id: "log-warning-injection-token",
+});
+
+export const logErrorInjectionToken = getInjectionToken<LogFunction>({
+  id: "log-error-injection-token",
+});
+
+export const logSillyInjectionToken = getInjectionToken<LogFunction>({
+  id: "log-silly-injection-token",
+});
```

Comment:
Also why are these categorically better?

Reply:
Depending from something that you don't use completely is considered as bad practice. (**Interface segregation principle**) E.g. if you depend from `logger` which is object containing multiple methods, you'll likely use only one of them. This means that e.g. in the unit tests you have to override stuff that you are not interested.